src/tests.rs

```diff
@@ -0,0 +1,26 @@
+use dotenv::dotenv;
+use std::env::var;
+
+use crate::{Cloudinary, upload::UploadOptions, UploadResult::{Success, Error}};
+
+#[tokio::test]
+async fn test_image_upload_from_url() {
+    dotenv().ok();
+    let api_secret = var("CLOUDINARY_API_SECRET").expect("enviroment variables not set");
+    let api_key = var("CLOUDINARY_API_KEY").expect("enviroment variables not set");
+    let cloud_name = var("CLOUDINARY_CLOUD_NAME").expect("enviroment variables not set");
+
+    let cloudinary = Cloudinary::new(api_key, cloud_name, api_secret);
+    let image_url = "https://www.pngall.com/wp-content/uploads/14/Gojo-PNG-Free-Image.png";
```

Comment:
Let's use something smaller for testing. Something like https://upload.wikimedia.org/wikipedia/commons/c/ca/1x1.png